modules/systemd/boot.nix

```diff
@@ -0,0 +1,109 @@
+# Copyright 2022-2023 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}: let
+
+  # Ghaf configuration flag
+  cfg = config.ghaf.systemd.boot;
+
+  # Import minimal systemd config
+  minimalSystemd = (import ./minimal.nix { inherit pkgs; }).minimalSystemd;
+
+  # Initrd-systemd package configuration 
+  bootPackageConfiguration =  minimalSystemd.override {
+    pname = "boot-systemd";
+    withAcl = true;
+    withCompression = true;
+    withCoredump = true;
+    withCryptsetup = true;
+    withEfi = pkgs.stdenv.hostPlatform.isEfi;
+    withFido2 = true;
+    withKmod = true;
+    withTpm2Tss = true;
+  };
+    
+  # Excluded systemd units for initrd
+  excludedInitrdUnits = [
+      
+    # Targets
+    # "basic.target"
+    # "swap.target"
+    # "initrd.target"
+    # "initrd-fs.target"
+    # "initrd-switch-root.target"
+    # "initrd-root-device.target"
+    # "initrd-root-fs.target"
+    # "sysinit.target"
+    (if config.ghaf.profiles.debug.enable then "emergency.target" else "")    
```

Comment:
It would be cleaner if you rewrite this way
```nix
[
  # targets
] ++ lib.optional config.ghaf.profiles.debug.enable "emergency.target";
```

Reply:
yes will do that - thanks